test/statefulset_storage_test.go

```diff
@@ -0,0 +1,202 @@
+package test
+
+import (
+	"strconv"
+	"time"
+
+	goctx "context"
+	asdbv1beta1 "github.com/aerospike/aerospike-kubernetes-operator/api/v1beta1"
+	"github.com/aerospike/aerospike-kubernetes-operator/pkg/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe(
+	"STSStorage", func() {
+		ctx := goctx.TODO()
+
+		Context(
+			"When doing valid operations", func() {
+
+				BeforeEach(
+					func() {
+						clusterName := "sts-storage"
+						clusterNamespacedName := getClusterNamespacedName(
+							clusterName, namespace,
+						)
+
+						aeroCluster := createDummyAerospikeCluster(
+							clusterNamespacedName, 2,
+						)
+
+						// Setup: Deploy cluster without rack
+						err := deployCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+				AfterEach(
+					func() {
+						clusterName := "sts-storage"
+						clusterNamespacedName := getClusterNamespacedName(
+							clusterName, namespace,
+						)
+
+						aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						err = deleteCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				It(
+					"Should allow external volume mount on aerospike nodes", func() {
+						clusterName := "sts-storage"
+						clusterNamespacedName := getClusterNamespacedName(
+							clusterName, namespace,
+						)
+
+						aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						sts, err := getSTSFromRackID(aeroCluster, 0)
+						Expect(err).ToNot(HaveOccurred())
+
+						mountedVolumes := sts.Spec.Template.Spec.Containers[0].VolumeMounts
+						mountedVolumes = append(mountedVolumes, v1.VolumeMount{
+							Name: "tzdata", MountPath: "/etc/localtime",
+						})
+
+						volumes := sts.Spec.Template.Spec.Volumes
+						volumes = append(volumes, v1.Volume{
+							Name: "tzdata",
+							VolumeSource: v1.VolumeSource{
+								HostPath: &v1.HostPathVolumeSource{
+									Path: "/etc/localtime",
+								},
+							},
+						})
+
+						sts.Spec.Template.Spec.Containers[0].VolumeMounts = mountedVolumes
+						sts.Spec.Template.Spec.Volumes = volumes
+
+						err = updateSTS(k8sClient, ctx, sts)
```

Comment:
AKO also updates the STS in reconcile. Are you sure it hasn't removed the tzdata volume that was added by this update?

Reply:
Modified the logic such that external volumes won't be affected with AKO reconciliation